edx/commerce-coordinator
# feat: Integrated openedx-filters and implemented a small PoC pipeline

**Description:** Initial proof-of-concept pipeline using [openedx-filters](https://github.com/openedx/openedx-filters)

**JIRA:** [[REV-2704] Hoist Order History: semi-synchronous internal communication mechanism POC & ADR - JIRA](https://openedx.atlassian.net/browse/REV-2704)

**Installation instructions:** 
- Activate commerce-coordinator virtualenv
- `make requirements`

**Testing instructions:**

1. `./manage.py runserver localhost:8140 --settings=commerce_coordinator.settings.
local`
2. Hit http://localhost:8140/demo_lms/demo_get_sample_data/
3. See some sample data
4. Edit pipeline in `commerce_coordinator/settings/base.py` to change the order of the steps, drop a step, etc.

**Author concerns:** 
- ~Still need to write ADR~ Added a draft

- Currently uses openedx-filters directly, more work will be needed to:
  - Add tracing
  - Integrate pipeline configuration with signal configs
  - Add configuration validation

- Currently doesn't demonstrate cross-app connections (I don't think that will be substantially different though)

- openedx-filters are somewhat complex and documentation is ...sparse
